### PR TITLE
Priorityq

### PR DESCRIPTION
**Pull request by Jakeand3rson**

### CODE REVIEW

File: priority_queue.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            self.first_item = self.last_item = None
+            return priority_item.val
+        # iterate through list to find highest priority item
+        current_item = self.first_item.next_item
+        while current_item is not None:
+            if priority_item.priority > current_item.priority:
+                priority_item = current_item
+                current_item = current_item.next_item
+            else:
+                current_item = current_item.next_item
```
Comment (cewing) [inline priority_queue.py:46]: There's a great deal of repetition between this code and the code in `peek` below.  You can do better than that.

Also, what's the time complexity of this implementation?  It's very complicated.

File: test_priority_queue.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+def test_peek_2():
+    queue = Queue()
+    queue.insert("Cabbage", 5)
+    queue.insert("mayo", 88)
+    queue.insert("Slaw", 4)
+    queue.insert("Village People", 7)
+    queue.insert("Coffee", 1)
+    assert queue.peek() == "Coffee"
+    queue.pop()
+    assert queue.peek() == ("Slaw")
```
Comment (cewing) [inline test_priority_queue.py:97]: I'd like to see a test demonstrating that for two items inserted with the same priority, order is preserved, even with other items between them of lower or higher priority.

**Comment by Jakeand3rson** [comment]
So I thought about what we had and how complicated it was, so I thought I would start over and go as simple as possible. And I remembered how simple the example for the really great stack.py assignment you showed us last week. So I decided to try and do the same kind of thing with this.

**Comment by cewing** [comment]
@Jakealope: I assume your comment here is about the new priorityQ class you created.  It is much more in line with what I was expecting you all to do.  There's a trick to getting it right that you still haven't quite got yet, but this is much much much closer to the mark.

The trick to the binary heap is that the items you insert have to be comparable.  You have to be able to say that `item a` is less than, greater than or equal to `item b`.  If you are putting items into a priority queue and you want to use a binheap that just calls some piece of code like `if a <= b:` then the items you put in must be able to be compared like that.  Can we make a class in Python such that different instances of the class can be compared with each-other using the standard operators in Python (`<`, `==`, `>`, etc.)?

Think about that a while and see if you don't see it.  If not, we'll be talking it over in class in the morning, so...

**Comment by Jakeand3rson** [comment]
I'll chew on that. I look forward to talking about it in the AM.

**Pull request merged**